### DISCUSSION TITLE
UnicodeDecodeError

### DISCUSSION THREAD

**Discussion by FauxPawDD** [mariush444/gmapIcons2osmand#1] [Q&A]
Hi mariush444, 
I was very happy to find your python script to convert the My Map Icons in a kmz-file into gpx. Thank you!
I tried using it on a kmz-file I downloaded from Google My Maps. The map is from Iceland and contains quite a few icons, colours, etc. 
Unfortunately, when I want to run it, it throws the following error: UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 51162: character maps to <undefined>
My guess is, this is due to the characters used in icelandic texts. But I might easily be mistaken. I will attach the kmz. Maybe you have a solution? 
Thank you
FauxPawD
[MyMap_Iceland.zip](https://github.com/mariush444/gmapIcons2osmand/files/9191835/MyMap_Iceland.zip)
D

**Comment by mariush444** [comment]
Hi
Good news that it could be usefull for someone other than me :)

Sorry but I can't help now because I'm during long jorney, till november, without acces to my dev. envitonment.

If it is unicode then it shouldn't be problem. 
In other case;
just try to change this letter to "normal/standard"
It should help as fast workaround.
I try to verify when I'll be back.

regards


-------- Wiadomość oryginalna --------
Od: ***@***.***
Wysłane: 26 lipca 2022 19:54:57 GMT+03:00
Do: ***@***.***, ***@***.***
DW: ***@***.***
Temat: [mariush444/gmapIcons2osmand] UnicodeDecodeError (Discussion #1)

Hi mariush444, 
I was very happy to find your python script to convert the My Map Icons in a kmz-file into gpx. Thank you!
I tried using it on a kmz-file I downloaded from Google My Maps. The map is from Iceland and contains quite a few icons, colours, etc. 
Unfortunately, when I want to run it, it throws the following error: UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 51162: character maps to <undefined>
My guess is, this is due to the characters used in icelandic texts. But I might easily be mistaken. I will attach the kmz. Maybe you have a solution? 
Thank you
FauxPawD
[MyMap_Iceland.zip](https://github.com/mariush444/gmapIcons2osmand/files/9191835/MyMap_Iceland.zip)
D

-- 
Reply to this email directly or view it on GitHub:
https://github.com/mariush444/gmapIcons2osmand/discussions/1

**Comment by mariush444** [comment]
so, the problem is that some items are not points but tracks with TrackTransparency.
I deleted:
```
Wanderung
Detour von Route 1 (twice)
Myvatn ggf. umrunden
oder doch die ostliche Seite...
Route 917
zu Aldeyjarfoss, dann F26 weiter nach Sueden?
FJALLABAK ROUTE Teil 1
FJALLABAK ROUTE, Teil 2
Durch Lavafeld?
Kaldidalur Hochlandstraße
FROM ARNARSTAPI TOWARDS THE GLACIER
ÖNDVERÐARNES
F35, möglichst noch August
F232, nicht die F233 fahren!
F210 bis Kaldaklofskvisl river crossing
Route von Hrossaborg nach Campground Möðrudal, Egilsstaðir, Island
Route von Búðakirkja nach 356, Island
Ergänzung zu Raudfoss
```
from the file and the gpx was created.

I'll check what I can do with "TrackTransparency"
[iceland4.zip](https://github.com/mariush444/gmapIcons2osmand/files/9974768/iceland4.zip)

**Comment by mariush444** [comment]
uncomment line 198, please then bingo 
[Iceland0.zip](https://github.com/mariush444/gmapIcons2osmand/files/9974846/Iceland0.zip)
